.github/workflows/golangci-test.yml

```
@@ -0,0 +1,32 @@
+name: golangci-test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Be a more specific like in `golangci-lint.yaml` to only run when Golang files are modified:
```suggestion
on:
    push:
        branches:
            - main
        paths:
            - '**.go'
    pull_request:
        branches:
            - main
        paths:
            - '**.go'
```

---

Changed the workflow to run only in case of changes to go files